docs/examples/sensormanagement/actionable_platform_example.py

```diff
@@ -0,0 +1,341 @@
+# associator, which, in this case, pair detections with predictions based on their distance.
+
+
+from stonesoup.hypothesiser.distance import DistanceHypothesiser
+from stonesoup.measures import Mahalanobis
+hypothesiser = DistanceHypothesiser(predictor, updater, measure=Mahalanobis(), missed_distance=5)
+
+from stonesoup.dataassociator.neighbour import GNNWith2DAssignment
+data_associator = GNNWith2DAssignment(hypothesiser)
+
+
+# %%
+# Running the Tracking Loop
+# -------------------------
+# Finally we can run the tracking loop.
+# 
+# At each timestep we use our sensor manager to generate the optimal actions for our sensor and
+# platform. When we call the sensor manager's :meth:`~.SensorManager.choose_actions` method, a few
+# things are happening in the background. For each actionable we control (including our actionable
+# platform), the actionable's :class:`~.ActionGenerator` is used to retrieve all possible
+# actions that our sensor or platform could take at that timestep. What happens next will depend
+# on the kind of sensor manager used. In our case we chose a :class:`~.BruteForceSensorManager`,
+# so every combination of actions between sensor and platform are considered, and each one is
+# evaluated by our reward function. The combination of actions resulting in the highest reward
+# will be returned, and we then move our actionables accordingly.
+# 
+# After moving our actionable objects, we take a measurement with our sensor, and update our track
+# depending on what we see.
+
+
+from stonesoup.sensor.sensor import Sensor
+import copy
+from collections import defaultdict
+sensor_history = defaultdict(dict)
+
+measurements = []
+for timestep in timesteps[1:]:
+    chosen_actions = sensormanager.choose_actions({track}, timestep)
+    for chosen_action in chosen_actions:
+        for actionable, actions in chosen_action.items():
+            actionable.add_actions(actions)
+            actionable.act(timestep)
+            if isinstance(actionable, Sensor):
+                measurement = actionable.measure({truth[timestep]},
+                                                 noise=True)
+    measurements.append(measurement)
+    hypotheses = data_associator.associate({track},
+                                           measurement,
+                                           timestep)
+    
+    sensor_history[timestep][sensor] = copy.deepcopy(sensor)
+    hypothesis = hypotheses[track]
+    
+    if hypothesis.measurement:
+        post = updater.update(hypothesis)
+        track.append(post)
+    else:
+        track.append(hypothesis.prediction)
+
+# %%
+# Plotting
+# --------
+# As we can see, the actionable platform is able to follow the target as it moves across the
+# action space, keeping it within the sensor's range.
+
+
+import plotly.graph_objects as go
+from stonesoup.functions import pol2cart
+
+plotter.plot_sensors(platform)
+plotter.plot_tracks(track, mapping=(0, 2))
```

Comment:
```suggestion
plotter.plot_tracks(track, mapping=(0, 2), particle=True)
```
Might be nice to plot the particles to show how the particle state changes over time

Reply:
Just tried this but after the first couple of steps the particle cloud was too small to be visible, so I haven't included for now.